spark-plugins/src/main/java/co/cask/hydrator/plugin/batch/spark/DecisionTreeTrainer.java

```diff
@@ -0,0 +1,197 @@
+    List<String> categoricalFeaturesList = Arrays.asList(fields);
+
+    for (final String field : fields) {
+      Map<Object, Long> map = input.map(new Function<StructuredRecord, Object>() {
+        @Override
+        public Object call(StructuredRecord structuredRecord) throws Exception {
+          Schema schema = structuredRecord.getSchema().getField(field).getSchema();
+          if (structuredRecord.get(field) == null && !schema.isNullable()) {
+            throw new IllegalArgumentException(String.format("null value found for non-nullable field %s", field));
+          }
+          Schema.Type type = schema.isNullable() ? schema.getNonNullable().getType() : schema.getType();
+          if (!(type.equals(Schema.Type.INT) || type.equals(Schema.Type.DOUBLE) || type.equals(Schema.Type.FLOAT) ||
+            type.equals(Schema.Type.LONG))) {
+            return structuredRecord.get(field);
+          } else {
+            return null;
+          }
+        }
+      }).distinct().zipWithIndex().collectAsMap();
+
+      if (!(map == null || (map.size() == 1 && map.keySet().contains(null)))) {
+        categoricalFeaturesMap.put(field, map);
+      }
+    }
+
+    JavaRDD<LabeledPoint> trainingData = input.map(new Function<StructuredRecord, LabeledPoint>() {
+      @Override
+      public LabeledPoint call(StructuredRecord record) throws Exception {
+        List<Double> doubles = new ArrayList<Double>();
+        Set<String> keys = categoricalFeaturesMap.keySet();
+        for (String field : fields) {
+          if (keys.contains(field)) {
+            doubles.add((categoricalFeaturesMap.get(field).get(record.get(field))).doubleValue());
+          } else {
+            doubles.add(new Double(record.get(field).toString()));
+          }
+        }
+        return new LabeledPoint((Double) record.get(config.predictionField), Vectors.dense(Doubles.toArray(doubles)));
+      }
+    });
+
+    trainingData.cache();
+    Set<String> keys = categoricalFeaturesMap.keySet();
+    for (String key : keys) {
+      categoricalFeaturesInfo.put(categoricalFeaturesList.indexOf(key), categoricalFeaturesMap.get(key).size());
+    }
+
+    config.maxDepth = config.maxDepth == null ? 10 : config.maxDepth;
+    config.maxBins = config.maxBins == null ? 100 : config.maxBins;
+    final DecisionTreeModel model = DecisionTree.trainRegressor(trainingData, categoricalFeaturesInfo, IMPURITY,
+                                                                config.maxDepth, config.maxBins);
+    // save the model to a file in the output FileSet
+    JavaSparkContext sparkContext = context.getSparkContext();
+    FileSet outputFS = context.getDataset(config.fileSetName);
+    model.save(JavaSparkContext.toSparkContext(sparkContext),
+               outputFS.getBaseLocation().append(config.path).toURI().getPath());
+
+  }
+
+  @Override
+  public void prepareRun(SparkPluginContext sparkPluginContext) throws Exception {
+    // no-op; no need to do anything
```

Comment:
SparkSink does not implement prepareRun and it's abstract in BatchConfigurable, so it has to be implemented here.
Since, Decision Tree does not need any job config to be set in prepare run, hence keeping this empty.